src/transactions/OperationFrame.h

```diff
@@ -42,13 +43,14 @@ class OperationFrame
     doCheckValidForSoroban(SorobanNetworkConfig const& networkConfig,
                            Config const& appConfig, uint32_t ledgerVersion,
                            OperationResult& res,
-                           DiagnosticEventBuffer* diagnosticEvents) const;
+                           DiagnosticEventManager& diagnosticEvents) const;
     virtual bool doCheckValid(uint32_t ledgerVersion,
                               OperationResult& res) const = 0;
-    virtual bool doApply(AppConnector& app, AbstractLedgerTxn& ltx,
-                         Hash const& sorobanBasePrngSeed, OperationResult& res,
-                         std::shared_ptr<SorobanTxData> sorobanData,
-                         OpEventManager& opEventManager) const = 0;
+    virtual bool
+    doApply(AppConnector& app, AbstractLedgerTxn& ltx,
+            Hash const& sorobanBasePrngSeed, OperationResult& res,
+            std::optional<RefundableFeeTracker>& refundableFeeTracker,
```

Comment:
It feels a little weird (or perhaps just tragic) to me that refundableFeeTracker is present as an optional argument in all cases even though it only applies to a single op (though I guess this is no different than sorobanData in the old code). I wonder if it'd be a little tidier to add an overload of `doApply` such that it's not even an argument on the `doApply` implementations in most of the op frames. Or whether this would be a waste of effort.

Reply:
That's true, though the same can be told about sorobanBasePrngSeed (actually refundable fee tracker is used in 3 ops, the seed is used in just 1). I think we can clean up both, but I would prefer to do that after we have figured out the parallel apply, as it also influences the apply interface.